ADAL/tests/unit/ios/ADEnrollmentGatewayTests.m

```diff
@@ -0,0 +1,109 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <XCTest/XCTest.h>
+#import "XCTestCase+TestHelperMethods.h"
+#import "ADEnrollmentGateway.h"
+#import <objc/runtime.h>
+
+@interface ADEnrollmentGatewayTests : ADTestCase
+
+@property NSString* testJSON;
+@property IMP originalAllEnrollmentIds;
+
+@end
+
+@implementation ADEnrollmentGatewayTests
+
+- (void)setUp
+{
+    [super setUp];
+
+    self.testJSON = [NSString stringWithFormat:
+        @"{\"enrollment_ids\": [\n"
+                                    "{\n"
+                                        "\"tid\" : \"fda5d5d9-17c3-4c29-9cf9-a27c3d3f03e1\",\n"
+                                        "\"oid\" : \"d3444455-mike-4271-b6ea-e499cc0cab46\",\n"
+                                        "\"unique_account_id\" : \"60406d5d-mike-41e1-aa70-e97501076a22\",\n"
+                                        "\"user_id\" : \"mike@contoso.com\",\n"
+                                        "\"enrollment_id\" : \"adf79e3f-mike-454d-9f0f-2299e76dbfd5\"\n"
+                                    "},\n"
+                                     "{\n"
+                                         "\"tid\" : \"fda5d5d9-17c3-4c29-9cf9-a27c3d3f03e1\",\n"
+                                         "\"oid\" : \"6eec576f-dave-416a-9c4a-536b178a194a\",\n"
+                                         "\"unique_account_id\" : \"1e4dd613-dave-4527-b50a-97aca38b57ba\",\n"
+                                         "\"user_id\" : \"dave@contoso.com\",\n"
+                                         "\"enrollment_id\" : \"64d0557f-dave-4193-b630-8491ffd3b180\"\n"
+                                     "},\n"
+                                    "]\n"
+            "}"];
+
+    // mock out enrollment data
+
+    Class ADEnrollmentGatewayClass = [ADEnrollmentGateway class];
+    SEL allEnrollmentIDsSelector = @selector(allEnrollmentIds);
+    Method allEnrollmentIDsMethod = class_getClassMethod(ADEnrollmentGatewayClass, allEnrollmentIDsSelector);
+    self.originalAllEnrollmentIds = method_getImplementation(allEnrollmentIDsMethod);
+
+    method_setImplementation(allEnrollmentIDsMethod, imp_implementationWithBlock(^NSString*(){return self.testJSON;}));
+
+
+}
+
+- (void)tearDown
+{
+    [super tearDown];
+
+    // restore mocked method
+
+    Class ADEnrollmentGatewayClass = [ADEnrollmentGateway class];
+    SEL allEnrollmentIDsSelector = @selector(allEnrollmentIds);
+    Method allEnrollmentIDsMethod = class_getClassMethod(ADEnrollmentGatewayClass, allEnrollmentIDsSelector);
+
+    method_setImplementation(allEnrollmentIDsMethod, self.originalAllEnrollmentIds);
+}
+
+- (void) testenrollmentIdForUserId
```

Comment:
please also add test cases for when enrollment ids dictionary is empty or corrupted